Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp

```diff
@@ -573,6 +578,110 @@ namespace AzToolsFramework
+                AZ_TracePrintf("JJS", "FolderPath = %s\n", folderPath.c_str());
+                bool connectedToAssetProcessor = false;
+                AzFramework::AssetSystemRequestBus::BroadcastResult(
+                    connectedToAssetProcessor, &AzFramework::AssetSystemRequestBus::Events::AssetProcessorIsReady);
+
+                if (connectedToAssetProcessor)
+                {
+                    auto entries = GetSelectedAssets();
+
+                    for (auto entry : entries)
+                    {
+                        using namespace AZ::IO;
+                        AZStd::string originalFname;
+                        AssetBrowserEntry* item = entry;
+                        Path fromPath = item->GetFullPath();
+                        PathView filename = fromPath.Filename();
+                        Path toPath(folderPath);
+                        toPath /= filename;
+                        AssetChangeReportRequest request(
+                            AZ::OSString(fromPath.c_str()), AZ::OSString(toPath.c_str()), AssetChangeReportRequest::ChangeType::CheckMove);
+                        AssetChangeReportResponse response;
+
+                        if (SendRequest(request, response))
+                        {
+                            AZStd::string message;
+                            for (int i = 0; i < response.m_lines.size(); ++i)
+                            {
+                                message += response.m_lines[i] + "\n";
+                            }
+
+                            if (message.size())
+                            {
+                                QMessageBox msgBox(this);
+                                msgBox.setWindowTitle("Before Move Asset Information");
+                                msgBox.setIcon(QMessageBox::Warning);
+                                msgBox.setText("The asset you are moving may be referenced in other assets.");
+                                msgBox.setInformativeText("More information can be found by pressing \"Show Details...\".");
+                                QAbstractButton* moveButton = (QAbstractButton*)msgBox.addButton("Move", QMessageBox::YesRole);
+                                msgBox.setStandardButtons(QMessageBox::Cancel);
+                                msgBox.setDefaultButton(QMessageBox::Yes);
+                                msgBox.setDetailedText(message.c_str());
+                                QSpacerItem* horizontalSpacer = new QSpacerItem(600, 0, QSizePolicy::Minimum, QSizePolicy::Expanding);
+                                QGridLayout* layout = (QGridLayout*)msgBox.layout();
+                                layout->addItem(horizontalSpacer, layout->rowCount(), 0, 1, layout->columnCount());
+                                msgBox.exec();
+
+                                if (msgBox.clickedButton() == moveButton)
+                                {
+                                    AssetChangeReportRequest request2(
```

Comment:
Is there a more descriptive name we could use besides `request2` ?

Reply:
Replaaced with moveRequest.